File: UCF HSPT Documents/2010/Solutions/ryan.cpp

#include <stdio.h>
#include <string.h>


int main()
{
   FILE *   infile;
   char     line[256];
   int      numTimes;
   char *   ch;
   int      numProper;

   // Open the file
   infile = fopen("ryan.in", "r");

   // Read the number of times Ryan should write the message
   fgets(line, sizeof(line), infile);
   sscanf(line, "%d", &numTimes);

   // Loop until we get a zero
   while (numTimes != 0)
   {
      // Get the first attempt
      fgets(line, sizeof(line), infile);
      if ( (ch = strchr(line, '\n')) != NULL)
         *ch = '\0';

      // Loop until we're done
      numProper = 0;
      while (strcmp(line, "Am I done yet?") != 0)
      {
         // See if this attempt is written correctly
         if (strcmp(line, "Greeks! Go home!") == 0)
             numProper++;

         // Get the next attempt
         fgets(line, sizeof(line), infile);
         if ( (ch = strchr(line, '\n')) != NULL)
            *ch = '\0';
      }

      // Output as appropriate
      if (numProper < numTimes)
         printf("You have %d left to go.\n", numTimes - numProper);
      else
         printf("You're done.  Now don't do it again.\n");

      // Read the number of times Ryan should write the message for the next
      // session
      fgets(line, sizeof(line), infile);
      sscanf(line, "%d", &numTimes);
   }

   // Close the file
   fclose(infile);
}

File: UCF HSPT Project/src/y2010/ryan.java
import java.util.*;
import java.io.*;


public class ryan
{

	//Our two constant strings to look out for.
	String sol = "Greeks! Go home!";
	String don = "Am I done yet?";

	public ryan(Scanner in) throws Exception
	{
		//Number of times we need to write the phrase.
		int times;
		
		//Keep looping until we see a zero
		while((times = in.nextInt()) != 0)
		{
			//The number of times we have already written
			int writ = 0;
		
			//An infinite loop, we break out when we find the end phrase
			for(;;)
			{
				String lin = in.nextLine();
				
				//If it is right, add one. == works here too.
				if(sol.equals(lin))
					writ++;
				else if(don.equals(lin))	//Else if it is the end string, quit
					break;
			}
			
			//If we didn't write enough times, print the difference
			if(writ < times)
				System.out.println("You have " + (times - writ) + " left to go.");
			else	//Else print done message.
				System.out.println("You're done.  Now don't do it again.");
		
		
		}
	
	}

	public static void main(String[] args) throws Exception
	{
		new ryan(new Scanner(new File("ryan.in")));	//I like using this trick instead
											// of making everything static.
	}
}

